lib/rules/no-wait-for-side-effects.ts

```diff
@@ -56,6 +57,27 @@ export default createTestingLibraryRule<Options, MessageIds>({
 			);
 		}
 
+		function isCallerThen(
+			node:
+				| TSESTree.AssignmentExpression
+				| TSESTree.BlockStatement
+				| TSESTree.CallExpression
+				| TSESTree.SequenceExpression
+		): boolean {
+			if (!node.parent) {
+				return false;
+			}
+
+			const callExpressionNode = node.parent.parent as TSESTree.CallExpression;
+
+			const test =
+				isMemberExpression(callExpressionNode.callee) &&
+				ASTUtils.isIdentifier(callExpressionNode.callee.property) &&
+				callExpressionNode.callee.property.name === 'then';
+
+			return test;
+		}
```

Comment:
I think you can reuse our [`hasChainedThen`](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/lib/node-utils/index.ts#L128) to simplify the chained then detection.

Reply:
Good point this too. I couldn't get to work with `hasChainedThen`, but `hasThenProperty` did the job.